server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1047 @@
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseBadRequest, HttpResponseNotFound
+from django.views.generic import View
+
+
+from pulp.common import tags
+from pulp.server.async.tasks import TaskResult
+from pulp.server.auth import authorization
+from pulp.server.db.model.criteria import Criteria
+from pulp.server.exceptions import (InvalidValue, MissingResource, MissingValue,
+                                    OperationPostponed, UnsupportedValue)
+from pulp.server.managers import factory
+from pulp.server.managers.consumer.applicability import (regenerate_applicability_for_consumers,
+                                                         retrieve_consumer_applicability)
+from pulp.server.managers.schedule.consumer import (UNIT_INSTALL_ACTION, UNIT_UNINSTALL_ACTION,
+                                                    UNIT_UPDATE_ACTION)
+from pulp.server.tasks import consumer as consumer_task
+from pulp.server.webservices.controllers.decorators import auth_required
+from pulp.server.webservices import serialization
+from pulp.server.webservices.views.util import (_ensure_input_encoding,
+                                                generate_json_response,
+                                                generate_json_response_with_pulp_encoder,
+                                                generate_redirect_response,
+                                                json_body_required,
+                                                json_body_allow_empty)
+
+
+def expand_consumers(options, consumers):
+    """
+    Expand a list of users based on the flag specified in the query parameters.
+    The _href is always added by the serialization function used.
+    Supported options:
+      details = include details
+      bindings - include bindings
+
+    :param options: The (expanding) options.
+    :type options: dict
+    :param consumers: A list of consumers
+    :type consumers: list
+
+    :return: A list of expanded consumers.
+    :rtype: list
+    """
+    details = (options.get('details', 'false')).lower()
+    bindings = (options.get('bindings', 'false')).lower()
+    if details != 'false':
+        bindings = 'true'
+    # add bindings
+    if bindings != 'false':
+        ids = [c['id'] for c in consumers]
+        manager = factory.consumer_bind_manager()
+        criteria = Criteria({'consumer_id': {'$in': ids}})
+        bindings = manager.find_by_criteria(criteria)
+        collated = {}
+        for b in bindings:
+            lst = collated.setdefault(b['consumer_id'], [])
+            lst.append(b)
+        for _consumer in consumers:
+            _consumer['bindings'] = \
+                [serialization.binding.serialize(b, False)
+                    for b in collated.get(_consumer['id'], [])]
```

Comment:
This block is a little awkward. I think this would look better:

    _consumer['bindings'] = [
        serialization.binding.serialize(b, False) for b in collated.get(_consumer['id'], [])
    ]